tests/udp_socket.rs

```diff
@@ -334,6 +334,14 @@ fn smoke_test_connected_udp_socket(mut socket1: UdpSocket, mut socket2: UdpSocke
     expect_read!(socket1.recv(&mut buf), DATA2);
     expect_read!(socket2.recv(&mut buf), DATA1);
 
+    checked_write!(socket1.send(DATA1));
+    expect_events(
```

Comment:
I think this is because we didn't hit `WouldBlock` before reading from socket2, which is a requirement for re-registering: https://github.com/tokio-rs/mio/blob/976f2354d0e8fbbb64fba3bf017d7131f9c369a0/src/sys/windows/mod.rs#L81-L86

Could you try adding a `socket2.recv` call that expects `WouldBlock` to test this guess?

Reply:
Hm yea good point I can try that.